starboard/shared/starboard/media/codec_util.cc

```diff
@@ -107,13 +108,102 @@ SbMediaAudioCodec GetAudioCodecFromString(const char* codec,
     return kSbMediaAudioCodecPcm;
   }
 #if SB_API_VERSION >= 15
-  if (strcmp(codec, "iamf") == 0 || strncmp(codec, "iamf.", 5) == 0) {
+  if (strcmp(codec, "iamf") == 0 || IsIamfMimeType(codec)) {
     return kSbMediaAudioCodecIamf;
   }
 #endif  // SB_API_VERSION >= 15
   return kSbMediaAudioCodecNone;
 }
 
+#if SB_API_VERSION >= 15
+bool IsIamfMimeType(std::string mime_type) {
+  // Reference: Immersive Audio Model and Formats;
+  //            v1.0.0
+  //            6.3. Codecs Parameter String
+  // (https://aomediacodec.github.io/iamf/v1.0.0-errata.html#codecsparameter)
+  if (mime_type.find("iamf") == std::string::npos) {
+    return false;
+  }
+
+  constexpr int kMaxIamfCodecIdLength =
```

Comment:
Although addition is a cheap operation, should we consider directly put the total number here, and then comment how it is calculated?